src/main/java/org/fcrepo/storage/ocfl/DefaultOcflObjectSession.java

```diff
@@ -142,30 +147,48 @@ public synchronized void writeResource(final ResourceHeaders headers, final Inpu
         final var contentPath = encode(paths.getContentFilePath());
         final var headerPath = encode(paths.getHeaderFilePath());
 
+        final var contentDst = createStagingPath(contentPath);
```

Comment:
Just wondering what the reason was for moving creation of the content staging path outside of the `if (content != null) {`? Creating an unused directory isn't super expensive, but does seem like it'd unnecessarily happen for all updates to RDF resources. I am guessing it is for the `safeDelete(contentDst);` call, maybe just initialize to null and then do a null check in `safeDelete`?